File: 01-02-2024/media-ponderada/MediaPonderada.java
import java.util.Scanner;

public class MediaPonderada {

    private static class Nota {

        private final float valor;
        private final int peso;

        public Nota(float valor, int peso) {
            this.valor = valor;
            this.peso = peso;
        }

        public float getValor() {
            return valor;
        }

        public int getPeso() {
            return peso;
        }

        public float getNota() {
            return peso * valor;
        }

    }

    public static void main(String[] args) {
        Nota[] notas = new Nota[3];

        try (Scanner scanner = new Scanner(System.in)) {
            for (int i = 0; i < 3; i++) {
                System.out.println("Nota " + (i + 1) + ": ");
                float nota = scanner.nextFloat();

                System.out.println("Peso da Nota " + (i + 1) + ": ");
                int peso = scanner.nextInt();

                notas[i] = new Nota(nota, peso);
            }
        } catch (Exception exception) {
            exception.printStackTrace();
        }

        float total = 0;
        int pesoTotal = 0;

        for (Nota nota : notas) {
            total += nota.getNota();
            pesoTotal += nota.getPeso();
        }

        System.out.println("Média total: " + (total / pesoTotal));
    }

}

File: 01-02-2024/media-ponderada/media_ponderada.c
#include <stdio.h>

/** 
 * Optei por uma abordagem diferente e não usar 6 variáveis (3 notas, 3 pesos)
 * assim, por mais que seja um pouco de overengineering, ficou "mais limpo" na minha opinião,
 * apesar de mais lento (por usar 2 loops), esse impacto é irrelevante, uma vez que são loops pequenos.
*/

typedef struct Nota {
    float valor;
    int peso;
} Nota;

int main() {
    Nota notas[3] = {};

    for (int i = 0; i < 3; i++) {
        float nota;
        int peso;

        printf("Insira a nota e seu peso (%dº): ", (i + 1));
        scanf("%f %d", &nota, &peso);

        Nota dados = {
            .peso = peso,
            .valor = nota
        };

        notas[i] = dados;
    }

    int pesoTotal = 0;
    float total = 0;

    for (int i = 0; i < 3; i++) {
        Nota nota = notas[i];

        pesoTotal += nota.peso;
        total += (nota.valor * nota.peso);
    }

    float media = total / pesoTotal;

    printf("Média total: %.2f\n", media);

    return 0;
}